### PR TITLE
Event refactoring

### PR DESCRIPTION
**Pull request by RobertHerhold**
Addresses #323 

- [x] `@Subscribe` --> `@Listener`
- [x] Refactor event class names (mostly done)
- [x] Refactor 'Working with Events' page
- [x] Docs for registering a class as an event listener (not dynamic)

### CODE REVIEW

**Comment by Spongy** [comment]
Setting up PR reference, please wait...

File: source/plugin/events.rst [hunk 1]
```diff
@@ -53,67 +53,99 @@ In addition, the class containing these methods must be registered with the even
...
+
+    game.getEventManager().registerListeners(this, new ExampleHandler());
+
+
+
+Dynamically Registering Event Handlers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some plugins (such as scripting plugins) may wish to dynamically register an event handler. In that case the event
+handler is not a method annotated with ``@Listener``, but rather a class implementing the ``EventHandler`` interface.
```
Comment (Zidane) [inline source/plugin/events.rst]: This is actually EventListener.

File: source/plugin/events.rst [hunk 2]
```diff
@@ -252,7 +284,7 @@ Example: Adding a Callback to Disable Explosions and Spawn an Arrow
 
 .. code-block:: java
 
-    @Subscribe
+    @Listener
     // final not required unless using an inner class that needs access to it.
     public void onExplosion(final ExplosionEvent event) {
```
Comment (Zidane) [inline source/plugin/events.rst]: Usually the format is onSomethingEvent or onSomethingEventAsPlayer (if one cause is gonna be checked)

onBreakBlockEvent

onBreakBlockEventAsPlayer

File: source/plugin/message-sinks.rst [hunk 3]
```diff
@@ -180,7 +180,7 @@ will be applied to any ``Player`` that joins the server.
     import org.spongepowered.api.text.sink.MessageSink;
     import org.spongepowered.api.text.sink.MessageSinks;
 
-    @Subscribe
+    @Listener
     public void playerJoin(PlayerJoinEvent event) {
```
Comment (Zidane) [inline source/plugin/message-sinks.rst]: Should be

onClientConnectionJoin(ClientConnectionEvent.Join event)

**Comment by RobertHerhold** [comment]
Going to merge this. If there are any other additions, they can be added in another PR

**Pull request merged**